hudi-client/src/main/java/org/apache/hudi/metrics/MetricsReporterFactory.java

```diff
@@ -48,6 +51,10 @@ public static MetricsReporter createReporter(HoodieWriteConfig config, MetricReg
       case DATADOG:
         reporter = new DatadogMetricsReporter(config, registry);
         break;
+      case USER_DEFINED:
```

Comment:
we would remove this, please refer to https://github.com/apache/hudi/blob/master/hudi-client/src/main/java/org/apache/hudi/index/HoodieIndex.java#L56

Reply:
Ok, look good